test/blocs/drive_attach_cubit_test.dart

```diff
@@ -68,7 +122,109 @@ void main() {
     );
 
     blocTest<DriveAttachCubit, DriveAttachState>(
-      'set form "${AppValidationMessage.driveNotFound}" error when no valid drive could be found',
+      'driveKey field is added to form when private drive id is used',
+      build: () => driveAttachCubit,
+      setUp: () {},
+      act: (bloc) {
+        bloc.form.value = {
+          'driveId': validPrivateDriveId,
+        };
+        bloc.form.updateValueAndValidity();
+      },
+      wait: Duration(milliseconds: 1000),
+      verify: (bloc) {
+        expect(bloc.form.contains('driveKey'), isTrue);
+      },
+    );
+    blocTest<DriveAttachCubit, DriveAttachState>(
+      'initializeForm on attaching private drive while logged out emits correct states',
+      build: () => driveAttachCubit,
+      setUp: () {
+        when(() => profileCubit.state).thenAnswer((_) => ProfilePromptAdd());
+      },
+      act: (bloc) async {
+        await bloc.initializeForm(
+          driveId: validPrivateDriveId,
+          driveKey: validPrivateDriveKey,
+          driveName: validDriveName,
+        );
+      },
+      expect: () => [DriveAttachPrivateNotLoggedIn()],
+    );
+
+    group('attach private drive while logged in', () {
+      final invalidDriveKeyBase64 = base64Encode(Uint8List(32));
+      final invalidDriveKey = SecretKey(
+        decodeBase64ToBytes(invalidDriveKeyBase64),
+      );
+      setUp(() async {
+        final keyBytes = Uint8List(32);
+        fillBytesWithSecureRandom(keyBytes);
+        final wallet = getTestWallet();
+        when(() => profileCubit.state).thenReturn(
+          ProfileLoggedIn(
+            username: '',
+            password: '123',
+            wallet: wallet,
+            cipherKey: SecretKey(keyBytes),
+            walletAddress: await wallet.getAddress(),
+            walletBalance: BigInt.one,
+          ),
+        );
+        when(() => arweave.getLatestDriveEntityWithId(
+              validPrivateDriveId,
+              invalidDriveKey,
+            )).thenAnswer((_) => Future.value(null));
+      });
+      blocTest<DriveAttachCubit, DriveAttachState>(
```

Comment:
Nit: missing newline
```suggestion
      });
      
      blocTest<DriveAttachCubit, DriveAttachState>(
```

Reply:
[Added](https://github.com/ardriveapp/ardrive-web/pull/376/commits/b93f655850a4c3114512f04e2019d2b6721727ad)